markdown-mode.el

```diff
@@ -6474,6 +6515,130 @@ a list."
 
 ;;; Movement ==================================================================
 
+;; This function was originally derived from `org-beginning-of-line' from org.el.
+(defun markdown-beginning-of-line (&optional n)
+  "Go to the beginning of the current visible line.
+
+If this is a headline, and `markdown-special-ctrl-a/e' is not nil
+or symbol `reversed', on the first attempt move to where the
+headline text hashes, and only move to beginning of line when the
+cursor is already before the hashes of the text of the headline.
+
+If `markdown-special-ctrl-a/e' is symbol `reversed' then go to
+the hashes of the text on the second attempt.
+
+With argument N not nil or 1, move forward N - 1 lines first."
+  (interactive "^p")
+  (let ((origin (point))
+        (special (pcase markdown-special-ctrl-a/e
+                   (`(,C-a . ,_) C-a) (_ markdown-special-ctrl-a/e)))
+        deactivate-mark)
+    ;; First move to a visible line.
+    (if (bound-and-true-p visual-line-mode)
```

Comment:
We can see `visual-line-mode` even if we launch emacs by `emacs -Q`. So I think `bound-and-true-p` is not necessary. In addition to that it was introduced at Emacs 24.4 and we don't support such an old version.

`org-mode` still uses `bound-and-true-p` for `visual-line-mode` I suppose org-mode has not updated the code for newer versions.

Reply:
@syohex Thank you for the review. I've fixed it.